File: clients/dart/lib/hhm_client.dart
class HhmClient {
  HhmClient(String baseUrl, {this.bearerToken})
      : baseUrl = Uri.parse(baseUrl.endsWith('/') ? baseUrl.substring(0, baseUrl.length - 1) : baseUrl);

  final Uri baseUrl;
  final String? bearerToken;

  Uri endpoint(String path) => baseUrl.resolve(path.startsWith('/') ? path.substring(1) : path);
  Uri get health => endpoint('/healthz');
  Uri get config => endpoint('/api/config');
}

File: clients/erlang/src/hhm_client.erl
-module(hhm_client).
-export([endpoint/2, health/1, config/1]).

endpoint(Base0, Path0) ->
    Base = string:trim(Base0, trailing, "/"),
    Path = string:trim(Path0, leading, "/"),
    Base ++ "/" ++ Path.
health(Base) -> endpoint(Base, "/healthz").
config(Base) -> endpoint(Base, "/api/config").
